heron/stmgr/tests/cpp/server/stmgr_unittest.cpp

```diff
@@ -1521,7 +1521,8 @@ TEST(StMgr, test_tmaster_restart_on_same_address) {
   // Note: Here we sleep longer compared to the previous test as we need
   // to tmasterClient could take upto 1 second (specified in test_heron_internals.yaml)
   // to retry connecting to tmaster.
-  sleep(3);
+  while (regular_stmgr->GetPhysicalPlan()->stmgrs(1).data_port() == common.stmgr_baseport_ + 1)
```

Comment:
It's hard to define a time here, b/c we never know if the message is lost or just delayed. I think relying on CI is fine, otherwise still have the same failure.